module/VuFind/src/VuFind/DigitalContent/Factory.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * Digital Content Connector Factory Class
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2014.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  DigitalContent
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Brent Palmer <brent-palmer@icpl.org> 
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     (Fixme)
+ */
+namespace VuFind\DigitalContent;
+
+use Zend\ServiceManager\ServiceManager;
+use Zend\Config\Config;
+use Zend\Session\SessionManager;
+
+/**
+ * Digital Content Connector Factory Class
+ *
+ * @category VuFind
+ * @package  DigitalContent
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Brent Palmer <brent-palmer@icpl.org>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:hierarchy_components Wiki
+ *
+ * @codeCoverageIgnore
+ */
+class Factory
+{
+    /**
+     * Factory for OverdriveConnector.
+     *
+     * @param Fixme Main Config     
+     * @param Fixme Record Config
+     * @param SessionManager Session Manager
+     * @param FIxme ILS Authenticator   
+     *
+     * @return OverdriveConnector
+     */
+    public static function getOverdriveConnector(ServiceManager $sm)
```

Comment:
I'm trying to move away from static factories, since using the new ZF factory interface makes factories more reusable. It's probably worth rewriting this accordingly. See https://github.com/vufind-org/vufind/blob/master/module/VuFind/src/VuFind/Content/AbstractAmazonFactory.php for an example of the new factory style.

Reply:
Ok, just to make sure I understand.  I will create a Vufind\DigitialContent\OverdriveConnectorFactory (or something similar) and implement the _invoke method, right?  Or will it be more generic than that?

Reply:
Yes, exactly right. One of the main benefits of this new approach is that you create `new $requestedName(...)` instead of using a hard-coded class name. Then, if somebody creates a subclass to customize something, they don't need to copy or extend the default factory unless they change the constructor signature -- one less thing to have to maintain in local custom code.